### PR TITLE
Move calling afterProperties on consumer endpoints from binder implem…

### PR DESCRIPTION
**Pull request by sobychacko**
…entations to core

### CODE REVIEW

File: spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/AbstractMessageChannelBinder.java [hunk 1]
```diff
@@ -179,6 +180,9 @@ protected void afterUnbindProducer(String destination, P producerProperties) {
 			bridge.setBeanName("bridge." + name);
 			consumerEndpoint = createConsumerEndpoint(name, group, destination, properties);
 			consumerEndpoint.setOutputChannel(bridge);
+			if(consumerEndpoint instanceof IntegrationObjectSupport) {
+				((IntegrationObjectSupport) consumerEndpoint).afterPropertiesSet();
```
Comment (mbogoevici) [inline spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/AbstractMessageChannelBinder.java]: This should be `InitializingBean`

Comment (mbogoevici) [inline spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/AbstractMessageChannelBinder.java]: Also, we need to do the same for the producer handler and documented the contract in the Javadoc.

**Comment by sobychacko** [comment]
@mbogoevici PR comments are addressed.

**Comment by mbogoevici** [comment]
Squashed and merged w/https://github.com/spring-cloud/spring-cloud-stream/commit/8dd22ebca0bd6439d3f4d75ce8674f07d09eca70

**Pull request closed without merging**